pkgs/development/interpreters/ruby/default.nix

```diff
@@ -59,6 +48,13 @@ let
       , buildEnv, bundler, bundix
       , libiconv, libobjc, libunwind, Foundation
       , makeWrapper, buildRubyGem, defaultGemConfig
+      , baseRuby ? if stdenv.hostPlatform != stdenv.buildPlatform
+        then buildPackages.ruby
+        else if useRailsExpress then self.override {
+          useRailsExpress = false;
+          docSupport = false;
+          rubygemsSupport = false;
+        } else null
```

Comment:
It would be nice if we wouldn't null this.

Reply:
what would you like as an alternative? a build can be performed with or without a `baseRuby`, it just felt naturally nullable to me.

Reply:
IIUC, the pattern Sandro prefers here is to always define baseRuby, and then have a separate bool-typed parameter (maybe useBaseRuby) that determines whether it should be used. I'll try to demonstrate with an inline suggestion, if that makes more sense.